src/transformStateWithClones.js

```diff
@@ -7,7 +7,48 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  const copyState = Object.assign({}, state);
+
+  for (const action of actions) {
+    transform(copyState, action);
+  }
+
+  function transform(newState, action) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(copyState, action.extraData);
```

Comment:
it's not good idea to use `dirty` function here
`dirty` because it's using variable outside a function for no reason